src/nanotron/trainer.py

```diff
@@ -578,14 +692,28 @@ def _load_model_checkpoint(self, model: NanotronModel) -> NanotronModel:
                     parallel_context=self.parallel_context,
                     root_folder=self.config.model.init_method.path,
                 )
+            elif isinstance(self.config.model.init_method, MambaInit):
```

Comment:
Since we put Mamba in /examples, we should decouple all Mamba-related logic from the Nanotron codebase and write a new Mamba trainer like this, for example https://github.com/huggingface/nanotron/blob/9f117eeb1e8309c45e2b1820069292ed5d7c4647/examples/doremi/doremi/trainer.py#L37